vms/evm/subscriber.go

```diff
@@ -154,25 +159,33 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Cap the number of blocks to process to MaxBlocksToProcess
-	toBlock := big.NewInt(0).SetUint64(latestBlock)
-	if height.Cmp(big.NewInt(0).Sub(toBlock, big.NewInt(MaxBlocksToProcess))) < 0 {
-		s.logger.Warn(
-			fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %d blocks", MaxBlocksToProcess),
-			zap.String("requestedBlockHeight", height.String()),
-			zap.String("latestBlockHeight", toBlock.String()),
-			zap.String("chainID", s.chainID.String()),
-		)
-		height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
+	bigLatestBlock := big.NewInt(0).SetUint64(latestBlock)
+
+	for fromBlock := big.NewInt(0).Set(height); fromBlock.Cmp(bigLatestBlock) <= 0; fromBlock.Add(fromBlock, big.NewInt(MaxBlocksPerRequest+1)) {
+		toBlock := big.NewInt(0).Add(fromBlock, big.NewInt(MaxBlocksPerRequest))
+		if toBlock.Cmp(bigLatestBlock) > 0 {
+			toBlock.Set(bigLatestBlock)
+		}
```

Comment:
Based on your findings discussed in https://github.com/ava-labs/awm-relayer/pull/89/files#r1402587975, it sounds this flooring is not strictly necessary. I understand the reason for keeping it, but can you leave a comment explaining so?

Reply:
addressed in https://github.com/ava-labs/awm-relayer/pull/89/commits/f967bdd8c80e1b06cba033330fd2f9a358513665